From 0442fd776c5cf7647b1da8ef84e9ecaf55d95b17 Mon Sep 17 00:00:00 2001
From: Arun Ravindran <arun.ravindran@intel.com>
Date: Mon, 11 Feb 2013 15:24:49 +0200
Subject: [PORT FROM MAIN] Telephony: Fix issue with overlapping stall alarms

BZ: 68905

When the data connection is disconnected due to NW DETACH, the
data stall alarm is not stopped. If the deactivation of context
after NW DETACH takes a longer time and there are active applications
trying sending and receiving, the data stall alarm is possible.
This alarm might disconnect an active connection if there is any
established after the deactivate of the first connection.

This patch stops the data stall alarm, if there are no data connections.

Change-Id: Idb636337e664612ce87b9b343510df9cab1a19a8
Orig-Change-Id: I6aaa9b35d9a93a9795ff7cdeef7b3de587648205
Category: aosp improvement
Domain: Telephony-Telephony-FWK
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../telephony/dataconnection/DcTracker.java        |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
index a9f46ba..803558c 100644
--- a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
+++ b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
@@ -1171,6 +1171,12 @@ public final class DcTracker extends DcTrackerBase {
             if (isAnyDataCallActive) startNetStatPoll();
         }
 
+        if (dataCallStates.size() == 0) {
+            if (DBG) log("onDataStateChange(ar): NoData calls Stop stall alarm");
+            stopNetStatPoll();
+            stopDataStallAlarm();
+        }
+
         if (DBG) log("onDataStateChanged(ar): X");
     }
 
-- 
1.7.4.1

